### DISCUSSION TITLE
useAnalyticsPageView(params) 에서 params변경시 pageview이벤트를 재호출하는 로직이 맞나?

### DISCUSSION THREAD

**Discussion by milooy** [EveryAnalytics/react-analytics-provider#155] [General]
첫 논의 스레드: https://www.notion.so/RAP-98df273e1c8b49d9be89107423a9425d#db3e8d475fc24d29a369b6e723a37e61

e.g.
```
const referrer = useQueryParam('referrer');
const [userNo, setUserNo] = useState('');
useAnalyticsPageView({referrer, userNo})
```

위 상황에서 referrer나 userNo가 바뀌면 PageView가 또 찍힙니다.

[대안]
params를 useEffect depths에서 제거
```
  React.useEffect(() => {
    analytics.onPageView(params);
  }, [analytics]); 
```

문제: useQueryParam등 꼭 필요한 인자가 비동기로 늦게 들어오면? (e.g. referrer: '' -> referrer: 'facebook')

useAnalyticsPageView에 함수도 넘길 수 있도록 해주면 어떠려나요

```
useAnalyticsPageView(() => {
  if (referrer === '') {
     return;
  }
  return {referrer}
})
```

의견 부탁스

**Comment by greatSumini** [comment]
params를 deps에서 제거해서 무조건 한번만 찍히고, 비동기는 콜백으로 처리하는 방식 좋은 것 같습니다 😄 

지금 params를 얕게 체크하고 있어서 발생할 수 있는 오작동도 함께 방지돼서 좋네요 ㅎㅎ

**Comment by MiryangJung** [comment]
useAnalyticsPageView에 함수도 넘길 수 있도록하는 방식 좋습니다!

**Comment by greatSumini** [comment]
resolved via #180
